miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/ads/AdMobDisplayer.kt

```diff
@@ -11,11 +12,24 @@ import com.google.android.gms.ads.rewarded.RewardItem
 import com.google.android.gms.ads.rewarded.RewardedAd
 import com.google.android.gms.ads.rewarded.RewardedAdCallback
 import com.google.android.gms.ads.rewarded.RewardedAdLoadCallback
+import com.rakuten.tech.mobile.admob.AdmobDisplayerLatest
+import com.rakuten.tech.mobile.admob.AdStatus
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlin.coroutines.CoroutineContext
 
+/** Check whether hostapp provides latest admob dependency. */
+@Suppress("EmptyCatchBlock", "SwallowedException")
+private inline fun <T> whenHasAdmobLatest(callback: (success: Boolean) -> T) {
```

Comment:
I wonder if it would make more sense to just make `AdmobDisplayerLatest` be it's own independent implementation of `MiniAppAdDisplayer`? It seems like this `AdMobDisplayer` class is becoming complicated (and you need to use reflection to check for `AdMobDisplayerLatest`).

But instead you could just implement the `MiniAppAdDisplayer` interface in `AdmobDisplayerLatest` and then if the App wants to use the latest Admob, then they should use `AdmobDisplayerLatest` instead of `AdMobDisplayer`. And then maybe we could rename `AdMobDisplayer` to `AdMob7Displayer` (and deprecate the old name using a `typealias`).

What do you think?

Reply:
@corycaywood The problem i faced is if you implement the latest admob sdk, the old version can not be used,  if we try to use it, it will create a crash, I think its android default behaviour, if multiple version of the same dependency detected it will automatically choose the latest one, so if we give an option to choose, host app might include the `admob-latest` module but try to implement old version, it will create a crash .. thats why i implemented the way that host app has to use the latest version if it include the `admob-latest` module in the gradle file , otherwise it will use the old sdk...does it make sense?

Reply:
yeah, if there is a crash / conflict between two libs, probably we need to handle the exception as well.

Reply:
@munir-rakuten I think it's correct that the App should crash because it means the Host App implemented something incorrectly. The Host App should know to use the current `AdMobDisplayer` if they're using Ads SDK 19 and use the new one from the module if they're using Ads SDK 20. As I mentioned, we can rename `AdMobDisplayer` to `AdMob19Displayer` and deprecate the old name:
```
@Deprecated("Use AdMobDisplayer19 if you're using Google Ads SDK 19. If you're using Google Ads SDK 20 or higher, use AdMobDisplyerLatest from the admob-latest module.")
typealias AdMobDisplayer = AdMobDisplayer19

class AdMobDisplayer19(private val context: Activity) : MiniAppAdDisplayer, CoroutineScope {
```